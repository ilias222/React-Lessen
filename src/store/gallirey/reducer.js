import { ADD_GALLIREY, GET_ARR_ERR, GET_ARR_SUCCES, GET_ARR_LOADING } from "./constant"

export const ChakTest = {
    Chak : [
    {
        names: 'Chak',
        id: 1,
        title: 'Im Chak Norris'
    },
    {
        names: 'Chak',
        id: 2,
        title: 'Im Big Boss'
    }
]
}

export const gallireyReducer = (state = ChakTest, action) => {
    const { type, payload } = action
    switch (type) {
        case ADD_GALLIREY:
            return {
                ...state,
                [payload.names]:[
                    ...state[payload.names],
                    {
                    names: payload.names, 
                    id: payload.id,
                    title: payload.title
                }]
            }
            case GET_ARR_ERR:
                return {
                    ...state,
                    [payload]:[
                        ...state[payload],
                        {
                        names: payload.error, 
                        id: payload.id,
                        date: payload.DATETYPE,
                        error: payload.err
                    }]
                }
                case GET_ARR_SUCCES:
                    return {
                        ...state,
                        [payload.name]:[
                            ...state[payload],
                            {
                            names: payload.succes, 
                            id: payload.id,
                            date: payload.DATETYPE
                        }]
                    }
                    case GET_ARR_LOADING:
                        return {
                            ...state,
                            [payload]:[
                                ...state[payload],
                                {
                                names: payload.loading, 
                                id: payload.id,
                                date: payload.DATETYPE,
                            }]
                        }
    
        default:
            return state
    }
}