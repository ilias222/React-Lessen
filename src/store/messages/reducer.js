import { ADD_CHAT, DELETE_CHAT, ADD_MESSAGE, ADD_MESSAGE_BOT } from './actions'
import { AUTHOR } from '../../constants'

const initailState = {
  default: [
    {
      author: 'user',
      text: 'one text'
    },
    {
      author: 'user',
      text: 'two text'
    },
  ]
}

export const messagesReducer = (state = initailState, action) => {
  const { type, payload } = action
  switch (type) {
    case ADD_CHAT:
      return {
        ...state,
        [payload]: []
      }

    case DELETE_CHAT:
      const chats = { ...state }
      delete chats[payload]
      return chats

    case ADD_MESSAGE:
      return {
        ...state,
        [payload.chatName]: [
          ...state[payload.chatName],
          {
            author: payload.name,
            text: payload.text
          }
      ],
      }

      case ADD_MESSAGE_BOT:
      return {
        ...state,
        [payload.chatName]: [
          ...state[payload.chatName],
          {
            author: AUTHOR.bot,
            text: payload.text
          }
        ],
      }

    default:
      return state
  }
}