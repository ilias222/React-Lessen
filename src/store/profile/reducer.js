import * as types from './types'

const initialState = {
  name: 'Goga',
  visible: true,
  isAuhc: false
}

export const profileReducer = (state = initialState, action) => {
  const { type, payload } = action
  switch (type) {
    case types.CHANGE_NAME:
      return {
        ...state,
        name: payload
      }
    case types.TOGGLE_PROFILE:
      return {
        ...state,
        visible: !state.visible
      }

      case types.ADD_AUCH:
      return {
        ...state,
        isAuhc: payload
      }

    default:
      return state
  }
}