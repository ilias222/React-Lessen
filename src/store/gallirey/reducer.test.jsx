import React from "react"
import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom'

import { gallireyReducer } from "./reducer"

describe('testing gallireyReducer', () => {
    it("returns state with status loading after requestArticles action",
        () => { 
       
            const initialState=null
                const userStore = gallireyReducer(initialState, {name: 'new Name'});
                expect(userStore).toMatchSnapshot();
                });
        })
