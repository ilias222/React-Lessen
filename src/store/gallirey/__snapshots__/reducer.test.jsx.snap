// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing gallireyReducer returns state with status loading after requestArticles action 1`] = `null`;
